src/witan/httpapi/components/activities.clj

```diff
@@ -160,17 +160,30 @@
                           {:partition-key file-id})
     (return-receipt id)))
 
+(defn- conform-metadata-updates
+  "This fn attempts to heal any differences between specs across the applications"
+  [m]
+  (log/debug "Conforming metadata updates:" m)
+  (let [m' (reduce-kv (fn [a k v]
+                        (assoc a k (s/unform k (s/conform k v)))) {} m)]
+    (log/debug "Conformed metadata updates:" m')
+    m'))
+
 (defn update-metadata!
   [{:keys [comms database]} user metadata-updates file-id]
-  (let [id (comms/uuid)]
-    (create-receipt! database user id)
-    (send-valid-command!* comms (merge {::command/id id
-                                        ::command/type :kixi.datastore.metadatastore/update
-                                        ::command/version "1.0.0"
-                                        :kixi/user user}
-                                       (assoc metadata-updates :kixi.datastore.metadatastore/id file-id))
-                          {:partition-key file-id})
-    (return-receipt id)))
+  (if (empty? metadata-updates)
+    [BAD_REQUEST {:error "Update can't be empty"} nil]
+    (let [id (comms/uuid)]
+      (create-receipt! database user id)
+      (send-valid-command!* comms (merge {::command/id id
+                                          ::command/type :kixi.datastore.metadatastore/update
+                                          ::command/version "1.0.0"
+                                          :kixi/user user}
+                                         (-> metadata-updates
```

Comment:
this bit in the `->` feels like something I'd extract into a function and have a unit or generative/spec based test for.